bin/update-ips.bash

```diff
@@ -143,368 +260,280 @@ process_ips_to_cidrs() {
+allow_object_storage() {
+  local cluster="${check_cluster}"
+  if [ "${check_cluster}" == "both" ]; then
+    cluster="sc"
+  fi
+
+  local url
+  local host
+  local port
+  url=$(yq_read "${cluster}" '.objectStorage.s3.regionEndpoint' "")
+  host=$(parse_url_host "${url}")
+  port=$(parse_url_port "${url}" 443)
 
-  echo "$swift_url"
+  allow_domain "${config["override_common"]}" '.networkPolicies.global.objectStorage.ips' "${host}"
+  allow_ports "${config["override_common"]}" '.networkPolicies.global.objectStorage.ports' "${port}"
 }
 
-if [ "${CHECK_CLUSTER}" == "both" ]; then
-  DIG_CLUSTER="sc"
-else
-  DIG_CLUSTER="wc"
-fi
-S3_ENDPOINT="$(yq_dig "${DIG_CLUSTER}" '.objectStorage.s3.regionEndpoint' '""' | sed 's/https\?:\/\///' | sed 's/[:\/].*//')"
-if [[ "${S3_ENDPOINT}" == "" ]]; then
-  log_error "No S3 endpoint found, check your common-config.yaml (or ${DIG_CLUSTER}-config.yaml)"
-  exit 1
-fi
-S3_PORT="$(yq_dig 'sc' '.objectStorage.s3.regionEndpoint' '""' | sed 's/https\?:\/\///' | sed 's/[A-Za-z.0-9-]*:\?//' | sed 's/\/.*//')"
-if [ -z "$S3_PORT" ]; then
-  S3_PORT="443"
-fi
+# Allow ingresses in the common network policy configuration.
+allow_ingress() {
+  local cluster="${check_cluster}"
+  if [ "${check_cluster}" == "both" ]; then
+    cluster="sc"
+  fi
 
-OPS_DOMAIN="$(yq_dig "${DIG_CLUSTER}" '.global.opsDomain' '""')"
-if [[ "${OPS_DOMAIN}" == "" ]]; then
-  log_error "No ops domain found, check your common-config.yaml (or ${DIG_CLUSTER}-config.yaml)"
-  exit 1
-fi
+  local base_domain
+  local ops_domain
+  base_domain="$(yq_read "${cluster}" '.global.baseDomain' "")"
+  ops_domain="$(yq_read "${cluster}" '.global.opsDomain' "")"
 
-BASE_DOMAIN="$(yq_dig "${DIG_CLUSTER}" '.global.baseDomain' '""')"
-if [[ "${BASE_DOMAIN}" == "" ]]; then
-  log_error "No base domain found, check your common-config.yaml (or ${DIG_CLUSTER}-config.yaml)"
-  exit 1
-fi
+  allow_domain "${config["override_common"]}" '.networkPolicies.global.scIngress.ips' "grafana.${ops_domain}"
+  allow_domain "${config["override_common"]}" '.networkPolicies.global.wcIngress.ips' "non-existing-subdomain.${base_domain}"
+}
 
-## Add object storage ips to common config
-checkIfDiffAndUpdateDNSIPs "${S3_ENDPOINT}" ".networkPolicies.global.objectStorage.ips" "${config["override_common"]}"
-## Add object storage port to common config
-checkIfDiffAndUpdatePorts ".networkPolicies.global.objectStorage.ports" "${config["override_common"]}" "$S3_PORT"
+# Allow Swift object storage in the sc network policy configuration.
+allow_swift() {
+  swift_enabled || return 0
 
-## Add sc ingress ips to common config
-checkIfDiffAndUpdateDNSIPs "grafana.${OPS_DOMAIN}" ".networkPolicies.global.scIngress.ips" "${config["override_common"]}"
+  local os_auth_url
+  local os_auth_host
+  local os_auth_port
+  os_auth_url="$(yq_read "sc" '.objectStorage.swift.authUrl' "")"
+  os_auth_host=$(parse_url_host "${os_auth_url}")
+  os_auth_port=$(parse_url_port "${os_auth_url}" 5000)
```

Comment:
Shouldn't the default port always be based on the protocol?

Reply:
That makes sense to me but with this PR my goal is only to refactor the code, not make any behavior changes. I realize I did stray from that goal when sorting IPs but I felt that was fine since the actual result is still going to be the same.

I'll add it to my list of follow-ups, together with https://github.com/elastisys/compliantkubernetes-apps/issues/1857 and your offline remark on `.networkPolicies.rcloneSync.destinationObjectStorageSwift` maintaining all the necessary config and not relying on `.networkPolicies.global.objectStorage`.

Reply:
For posterity:

> Shouldn't the default port always be based on the protocol?

https://github.com/elastisys/compliantkubernetes-apps/pull/1876

> and your offline remark on `.networkPolicies.rcloneSync.destinationObjectStorageSwift` maintaining all the necessary config and not relying on `.networkPolicies.global.objectStorage`.

https://github.com/elastisys/compliantkubernetes-apps/pull/1873